packages/core/functions/src/handler.ts

```diff
@@ -22,13 +23,28 @@ import { isNonNullable } from '@dxos/util';
 /**
  * Function handler.
  */
-export type FunctionHandler<TData = {}, TMeta = {}, TOutput = any> = (params: {
+export type FunctionHandler<TData = {}, TOutput = any> = (params: {
+  /**
+   * Services and context available to the function.
+   */
   context: FunctionContext;
-  event: FunctionEvent<TData, TMeta>;
+
   /**
-   * @deprecated
+   * Trigger that invoked the function.
+   * In case the function is part of a workflow, this will be the trigger for the overall workflow.
    */
-  response: FunctionResponse;
+  trigger: FunctionTrigger;
+
+  /**
+   * Data passed as the input to the function.
+   * Must match the function's input schema.
+   */
+  data: TData;
+
+  /**
+   * Event data from the trigger.
+   */
+  event: EventType;
```

Comment:
in a workflow does this also come from the trigger for the overall workflow?

Reply:
yes